app/code/local/Magmodules/Webwinkelconnect/Model/Api.php

```diff
@@ -73,63 +73,76 @@ public function getFeed($storeid, $type = '')
-        $apiKey = trim(Mage::getStoreConfig('webwinkelconnect/general/api_key', $storeId));
-        $postData['email'] = $order->getCustomerEmail();
-        $postData['order'] = $order->getIncrementId();
-        $postData['delay'] = trim(Mage::getStoreConfig('webwinkelconnect/invitation/delay', $storeId));
-        $postData['customer_name'] = $order->getCustomerName();
-        $postData['client'] = 'magento';
-        $postData['language'] = $this->getLanguage($storeId, $order);
+        $start_time = microtime(true);
+        $store_id = $order->getStoreId();
+        $helper = Mage::helper('webwinkelconnect');
+        $post_data['email'] = $order->getCustomerEmail();
+        $post_data['order'] = $order->getIncrementId();
+        $post_data['delay'] = trim(Mage::getStoreConfig('webwinkelconnect/invitation/delay', $store_id));
+        $post_data['customer_name'] = $order->getCustomerName();
+        $post_data['client'] = 'magento';
+        $post_data['language'] = $this->getLanguage($store_id, $order);
 
         if (Mage::getStoreConfig('webwinkelconnect/product_review_invites/enabled')) {
-            $postData['order_data'] = json_encode([
+            $post_data['order_data'] = json_encode([
                 'order' => $order,
                 'products' => $this->getOrderProducts($order),
             ]);
         }
 
         $url = 'https://dashboard.webwinkelkeur.nl/api/1.0/invitations.json?' .
             http_build_query([
-                'id' => $apiId,
-                'code' => $apiKey,
+                'id' => $helper->getShopId(),
+                'code' => $helper->getApiKey(),
             ]);
 
-        $curl = new Varien_Http_Adapter_Curl();
-        $curl->setConfig([
-            'timeout'   => 10,
-        ]);
-        $curl->addOptions([
-            CURLOPT_RETURNTRANSFER => true,
-            CURLOPT_FOLLOWLOCATION => true,
-            CURLOPT_FAILONERROR => true,
-        ]);
-        $curl->write(Zend_Http_Client::POST, $url, '2', false, $postData);
-        $response = $curl->read();
-
-        if ($response) {
-            $responseHtml = $response;
-        } else {
-            $responseHtml = sprintf('(%s) %s',$curl->getErrno(), $curl->getError());
+        if (
+            Mage::getStoreConfig('webwinkelconnect/privacy_first_option/privacy_popup')
+            && !Mage::helper('webwinkelconnect')->hasConsent($order)
+        ) {
+            return false;
+        }
+
+        $ch = curl_init();
+        curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_FAILONERROR, true);
+        curl_setopt($ch, CURLOPT_POST, true);
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $post_data);
+        curl_setopt($ch, CURLOPT_URL, $url);
+        try {
+            $response = curl_exec($ch);
```

Comment:
Same thing about error handling. You could also add a helper function that takes the URL and cURL options (at least those that are not the same for every request), and throws exceptions if something goes wrong, so that you don't have to repeat this logic wherever you do a HTTP request.